src/index.ts

```diff
@@ -275,7 +279,12 @@ export function register (options: Options = {}): Register {
             return undefined
           }
 
-          cache.contents[fileName] = getFile(fileName)
+          const content = getFile(fileName)
+          if (shouldVerifyCache && !isValidCacheContent(content)) {
```

Comment:
Right, I see. Yeah, the input files are cached because they are read multiple times by TypeScript. I would love to improve this, but it's possible to end up inconsistent since the node.js loader can generate a different file _technically_ than the original source file. But the initial source file is required for TypeScript to work (importing types). Maybe it's worth experimenting whether we can go through `require` here instead, but I have a feeling that will fail with anything that needs to go _after_ `ts-node` to work such as code coverage.